comet/broker.py

```diff
@@ -198,9 +132,9 @@ def float_to_datetime(fl):
     logger.debug('send-state: Received {} state {}'.format(type, hash))
     reply = dict()
 
-    # do we have this state already?
-    async with lock_states:
-        found = states.get(hash)
+    # Lock states and check if we know this state already.
```

Comment:
Again, I think a redis script could probably do most of this and remove the need for the locking.

The disadvantage in this instance is if the JSON dump is large, you would always be sending it regardless of whether the state already exists. However, unless something breaks in the logic above, I think you should never be in this call if the state already exists.